File: wallet-management-system/src/main/java/com/walletsystem/wallet_management_system/account/service/AccountService.java
package com.walletsystem.wallet_management_system.account.service;

import com.walletsystem.wallet_management_system.account.dto.AccountRequest;
import com.walletsystem.wallet_management_system.account.dto.AccountResponse;

import java.util.List;

public interface AccountService {
    List<AccountResponse> getAllAccounts();
    List<AccountResponse> getAccountsByUserId(Long userId);
    AccountResponse getAccountById(Long id);
    AccountResponse createAccount(AccountRequest request);
    AccountResponse updateAccount(Long id, AccountRequest request);
    void deleteAccount(Long id);
}

File: wallet-frontend/src/services/accountService.js
import api from './api';

/**
 * Account Service
 * Handles all account CRUD operations
 */

// Get all accounts
export const getAccounts = async () => {
  const response = await api.get('/api/accounts');
  return response.data;
};

// Get accounts by user ID
export const getAccountsByUserId = async (userId) => {
  const response = await api.get(`/api/accounts/user/${userId}`);
  return response.data;
};

// Get account by ID
export const getAccountById = async (id) => {
  const response = await api.get(`/api/accounts/${id}`);
  return response.data;
};

// Create new account
export const createAccount = async (accountData) => {
  const response = await api.post('/api/accounts', accountData);
  return response.data;
};

// Update account
export const updateAccount = async (id, accountData) => {
  const response = await api.put(`/api/accounts/${id}`, accountData);
  return response.data;
};

// Delete account
export const deleteAccount = async (id) => {
  const response = await api.delete(`/api/accounts/${id}`);
  return response.data;
};

export default {
  getAccounts,
  getAccountsByUserId,
  getAccountById,
  createAccount,
  updateAccount,
  deleteAccount,
};
